### ISSUE TITLE
Due to Migration to 1.12 makes it difficult to run and debug this app

### ISSUE THREAD

**Issue by f2ka07** [FirebaseExtended/codelab-friendlyeats-flutter#21]
The migration is a bit complex as this app misses some of the files where changes that affect the migration are made

**Comment by ditman** [comment]
Yes, there's a new codelab with similar content that is more up to date:

https://firebase.google.com/codelabs/firebase-get-to-know-flutter#0

**Comment by JLT-98** [comment]
Hi Sir, after migrate the app
![image](https://user-images.githubusercontent.com/70704633/163657653-a55975bc-6ea1-486f-93b0-7d9a39231b57.png)

getting error of
The argument type 'void Function(T)?' can't be assigned to the parameter type 'void Function(T?)?'.
on the filterselectdialog.dart

what should i change for the 
typedef FilterChangedCallback<T> = void Function(T newValue)?;  from data.dart(codelab)

or what is the solution for the >>>The argument type 'void Function(T)?' can't be assigned to the parameter type 'void Function(T?)?'.

**Comment by JLT-Labs** [comment]
Can use this
https://github.com/ryokosuge/friendlyeats-flutter
https://github.com/reiko-dev/restaurant_recommendation
